include/swift/AST/DiagnosticsSema.def

```diff
@@ -2378,6 +2378,9 @@ NOTE(dependency_as_imported_by_main_module,none,
      "a dependency of main module '%0'", (StringRef))
 NOTE(dependency_as_imported_by, none,
      "a dependency of %select{Swift|Clang}2 module '%0': '%1'", (StringRef, StringRef, bool))
+GROUPED_NOTE(inherited_search_path_resolves_module,MissingModuleOnKnownPaths,none,
```

Comment:
I would expect the `MissingModuleOnKnownPaths` group to be associated with the original warning/error, rather than with the note. I see why you structured this way, but I think it could be done instead by emitting a different error diagnostic ID when the conditions are met